Functions/xASL_Iteration.m

```diff
@@ -328,11 +328,18 @@
         end
         
         % Some feedback about this iteration (after opening diary log)
+                
         if ~AlreadyProcessed
             fprintf('\n%s\n',repmat('+',1,72)); % just draw a separator line
-            fprintf('%s\n',['Starting ' x.MUTEXID ' at ' datestr(now)]);
+            [StartI, EndI] = regexp(diaryFileEx, 'ASL_\d','ONCE'); %to find the name of the session inside diaryFileEx: ASL_with any digit after
+            session= diaryFileEx(StartI:EndI); %isolate "ASL_1 or ASL_2 etc
```

Comment:
This will crash if `StartI` is empty. You could move it after the next line to ensure that this is only run if it isn't empty.
Also good to add a space between `session` and `=` for readability